src/reconcile/rec_write.c

```diff
@@ -638,10 +638,10 @@ __rec_init(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags, WT_SALVAGE_COO
 
     /*
      * The checkpoint transaction doesn't pin the oldest txn id, therefore the global last_running
-     * can move beyond the checkpoint transaction id. When reconciling the metadata, we have to take
-     * checkpoints into account.
+     * can move beyond the checkpoint transaction id. When reconciling the metadata or disaggregated
+     * shared metadata, we have to take checkpoints into account.
```

Comment:
Is this because checkpoint itself writes updates into these tables? Or something else?

Reply:
When we move the last running field, we don't consider the checkpoint id. If we evict based on last running to do eviction, the uncommitted checkpoint transaction can be evicted by eviction threads as it is smaller than last running.

Reply:
[0022d36](https://github.com/wiredtiger/wiredtiger/pull/11438/commits/0022d369c0972d37d7427cd0f84d4228a068f6df)